lib/phoenix_component.ex

```diff
@@ -573,6 +578,32 @@ defmodule Phoenix.Component do
 
   This list may increase in the future.
 
+  ## Documentation
+
+  Public function components that define attributes will have their attribute 
+  types and docs injected into the function's documentation, depending on the 
+  value of the `@doc` module attribute:
+
+    * if `@doc` is a string, the attribute docs are injected into that string. 
+      The optional placeholder `[[INJECT ATTRDOCS]]` can be used to specify where 
+      in the string the docs are injected. Otherwise, the docs are appended 
+      to the end of the `@doc` string.
+      
+    * if `@doc` is unspecified, the attribute docs are used as the 
+      default `@doc` string.
+      
+    * if `@doc` is false, the attribute docs are omitted entirely.
+    
+  The injected attribute docs are formatted as a markdown list:
+
+    ```markdown
+    * *name* _`:type, opts`_, doc
```

Comment:
Just one nitpick on the format. I think the `name` should be in backticks too. Requires can be a boolean too. We typically write the default in text (but either way is fine by me). So I would go with something like this:

  * `name` (`:any`) (required) - Yada yada yada. Defaults to `false`.

WDYT?

Reply:
I like this format, especially if it is consistent with existing libraries 😄